AppfulTests/TokenTest.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  TokenTest.swift
+//  Appful
+//
+//  Created by Karl Rivest Harnois on 16-04-28.
+//  Copyright © 2016 Dynamo. All rights reserved.
+//
+
+import XCTest
+import Appful
+
+class TokenTest: XCTestCase {
+    
+    let testEmail = "hugo@godynamo.com"
+    let testPassword = "123456"
+    let testToken = "9SNq/xFu/IPITXwQidPugg=="
+    
+    func testFetchToken(){
+        let fetch = fetchToken(testEmail, testPassword) { data, response, error in
+            let statusCode = (response as! NSHTTPURLResponse).statusCode
+            XCTAssertTrue(statusCode == 200, "fetchToken() status code should be 200 for test email/password")
+            XCTAssertTrue(data != nil, "fetchToken() should return token data for test email/password")
+            XCTAssertTrue(self.testToken == token(data!), "fetchToken() data should return testToken when applying token function")
+            XCTAssertNil(error, "fetchToken() error should be nil")
```

Comment:
That make senses, I'll do that with my next tests thanks!


________________________________
De : Hugo Bastien <notifications@github.com>
Envoyé : 29 avril 2016 07:59
À : hugobast/Appful
Cc : Karl Rivest Harnois; Mention
Objet : Re: [hugobast/Appful] Authentication (#2)


In AppfulTests/TokenTest.swift<https://github.com/hugobast/Appful/pull/2#discussion_r61566223>:

> +import XCTest
> +import Appful
> +
> +class TokenTest: XCTestCase {
> +
> +    let testEmail = "hugo@godynamo.com"
> +    let testPassword = "123456"
> +    let testToken = "9SNq/xFu/IPITXwQidPugg=="
> +
> +    func testFetchToken(){
> +        let fetch = fetchToken(testEmail, testPassword) { data, response, error in
> +            let statusCode = (response as! NSHTTPURLResponse).statusCode
> +            XCTAssertTrue(statusCode == 200, "fetchToken() status code should be 200 for test email/password")
> +            XCTAssertTrue(data != nil, "fetchToken() should return token data for test email/password")
> +            XCTAssertTrue(self.testToken == token(data!), "fetchToken() data should return testToken when applying token function")
> +            XCTAssertNil(error, "fetchToken() error should be nil")


General comment: I try to keep my tests as one assert per test so that when XCTAssertTrue(statusCode == 200) fails for example because we changed it to a 204 the rest of my assertions will still get run.

-
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/hugobast/Appful/pull/2/files/29ae6e7d061f2455f404dd97ea2adab2967e492d#r61566223>